config/schema_helpers.go

```diff
@@ -91,5 +114,37 @@ func parseValidTypesFromSchema(schema map[string]interface{}, context string) []
 		}
 	}
 
+	if _, ok := property["oneOf"]; !ok {
+		key = contextSplit[len(contextSplit)-2]
+
+		property := make(map[string]interface{})
+		if _, ok := properties[key]; ok {
+			property = properties[key].(map[string]interface{})
+		}
+
+		if _, ok := property["patternProperties"]; ok {
+			// address schema for 'ulimits'
+			patternProperties := make(map[string]interface{})
+			patternProperties = property["patternProperties"].(map[string]interface{})
+
+			regex := make(map[string]interface{})
+			if _, ok := patternProperties["^[a-z]+$"]; ok {
```

Comment:
I don't think we should hardcode this pattern here. I think you can generalize it to something like (brace for bad psuedo-code):
```
for pattern, patternVal := range patternProperties {
  if val, ok patternVal["oneOf"]; ok {
    validTypes = oneOfValidTypes(val, validTypes)
  }
}
```